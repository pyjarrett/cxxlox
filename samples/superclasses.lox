class Parent {
    parentMethod() {
        print "Parent method";
    }

    name() {
        return "Parent";
    }
}

class Child < Parent {
    childMethod() {
        print "Child method";
    }

    parentName() {
        return super.name();
    }

    name() {
        return "Child";
    }
}

var child = Child();
child.childMethod();
child.parentMethod();
print child.parentName();