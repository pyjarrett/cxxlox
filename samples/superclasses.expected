Child method
Parent method
Parent
